pkg/generate/code/equal.go

```diff
@@ -0,0 +1,263 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package code
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	awssdkmodel "github.com/aws/aws-sdk-go/private/model/api"
+
+	ackmodel "github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/aws-controllers-k8s/code-generator/pkg/names"
+)
+
+// IsEqualTypeDef returns Go code that checks the equality of two struct types.
+// The generated code returns false at the first spotted difference.
+//
+// Output code will look something like this:
+//
+//  if ackcompare.HasNilDifference(a.CreatedAt, b.CreatedAt) {
+//  	return false
+//  } else if a.CreatedAt != nil && b.CreatedAt != nil {
+//  	if *a.CreatedAt != *b.CreatedAt {
+//  		return false
+//  	}
+//  }
+//  if ackcompare.HasNilDifference(a.EncryptionConfiguration, b.EncryptionConfiguration) {
+//  	return false
+//  } else if a.EncryptionConfiguration != nil && b.EncryptionConfiguration != nil {
+//  	if !EncryptionConfiguration(a.EncryptionConfiguration, b.EncryptionConfiguration) {
```

Comment:
The above would be `if !IsEqualEncryptionConfiguration(...) {`, right?

Reply:
Yepp ++